File: Music.Client.Android/Java/music/app/src/main/java/com/zitherharp/store/Repository.java
package com.zitherharp.store;

import androidx.annotation.NonNull;

import com.zitherharp.store.model.Application;

import org.jetbrains.annotations.NotNull;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.net.URL;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Repository {
    @NonNull
    public static List<Application> getApplications() {
        Application application;
        JSONArray jsonValue, jsonValues = getJSONArray();
        List<Application> applications = new ArrayList<>();
        for (int i = 0; i < jsonValues.length(); i++) {
            jsonValue = getJSONArray(jsonValues, i);
            application = new Application();
            application.id = getString(jsonValue, 0);
            application.iconId = getString(jsonValue, 1);
            application.name = getString(jsonValue, 2);
            application.description = getString(jsonValue, 3);
            application.version = getString(jsonValue, 4);
            application.size = getInt(jsonValue, 5);
            application.link = getString(jsonValue, 6);
        }
        return applications;
    }

    private static JSONArray getJSONArray() {
        JSONArray jsonArray = null;
        String url = "https://sheets.googleapis.com/v4/spreadsheets/" +
                "1znQOtTDJz0UqDs0uB2MQZV3wN0l_J0TrU44d9chH2SI/values/" +
                "application?key=AIzaSyAD91OiEeWRoqhsw0peq94qg5joZe47r_s";
        try {
            Scanner scanner = new Scanner(new URL(url).openStream());
            StringBuilder jsonString = new StringBuilder(scanner.next());
            while (scanner.hasNextLine()) {
                jsonString.append(scanner.nextLine());
            }
            JSONObject jsonObj = new JSONObject(jsonString.toString());
            jsonArray = jsonObj.getJSONArray("values");
        } catch (IOException | JSONException e) {
            e.printStackTrace();
        }
        return jsonArray;
    }

    private static JSONArray getJSONArray(@NotNull JSONArray jsonArray, int index) {
        JSONArray valueArray = null;
        try {
            valueArray = jsonArray.getJSONArray(index);
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return valueArray;
    }

    private static String getString(@NotNull JSONArray jsonArray, int index) {
        String string = "";
        try {
            if (!jsonArray.isNull(index)) {
                string = jsonArray.getString(index);
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return string;
    }

    private static int getInt(@NotNull JSONArray jsonArray, int index) {
        int integer = 0;
        try {
            if (!jsonArray.isNull(index)) {
                integer = jsonArray.getInt(index);
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return integer;
    }
}

File: Music.Client.Android/Kotlin/Store/app/src/main/java/com/zitherharp/store/Repository.kt
package com.zitherharp.store

import com.zitherharp.store.model.Item
import org.json.JSONArray
import org.json.JSONObject
import java.net.URL

object Repository {
    val itemMap: MutableMap<String, Item> = HashMap()

    init {
        var jsonValue: JSONArray
        // TODO: Load items
        var jsonValues = getJsonValues("item")
        for (i in 0 until jsonValues.length()) {
            jsonValue = jsonValues.getJSONArray(i)
            itemMap[jsonValue.requireString(0)] = Item(
                jsonValue.requireString(0),
                jsonValue.requireString(1),
                jsonValue.requireString(2),
                jsonValue.requireString(3),
                jsonValue.requireString(4))
        }
        // TODO: Load field
        jsonValues = getJsonValues("field")
        for (i in 0 until jsonValues.length()) {
            jsonValue = jsonValues.getJSONArray(i)
            itemMap[jsonValue.requireString(0)]?.let {
                it.field = Item.Field(
                    jsonValue.requireString(1),
                    jsonValue.requireDouble(2),
                    jsonValue.requireString(3),
                    jsonValue.requireInt(4),
                    jsonValue.requireString(5),
                    jsonValue.requireString(6))
            }
        }
        // TODO: Load image
        jsonValues = getJsonValues("image")
        for (i in 0 until jsonValues.length()) {
            jsonValue = jsonValues.getJSONArray(i)
            itemMap[jsonValue.requireString(0)]?.let {
                it.image = Item.Image(
                    jsonValue.requireString(1),
                    jsonValue.requireString(2),
                    jsonValue.requireString(3))
            }
        }
        // TODO: Load contact
        jsonValues = getJsonValues("contact")
        for (i in 0 until jsonValues.length()) {
            jsonValue = jsonValues.getJSONArray(i)
            itemMap[jsonValue.requireString(0)]?.let {
                it.contact = Item.Contact(
                    jsonValue.requireString(1),
                    jsonValue.requireString(2),
                    jsonValue.requireString(3),
                    jsonValue.requireString(4))
            }
        }
    }

    private fun getJsonValues(name: String): JSONArray {
        val url = "https://sheets.googleapis.com/v4/spreadsheets/" +
                "1jjImV0A_U9ecTzPfdoxJmc3TxiMr3YMb3qEeAcCmu1g/values/" +
                "$name?key=AIzaSyAD91OiEeWRoqhsw0peq94qg5joZe47r_s"
        val jsonString = URL(url).readText()
        val jsonObject = JSONObject(jsonString)
        return jsonObject.getJSONArray("values")
    }

    private fun JSONArray.requireString(index: Int) = try { getString(index) } catch (e: Exception) { "" }
    private fun JSONArray.requireInt(index: Int) = try { getInt(index) } catch (e: Exception) { 0 }
    private fun JSONArray.requireDouble(index: Int) = try { getDouble(index) } catch (e: Exception) { 0.0 }
}